java/kafka/producer/src/main/java/KafkaProducerExample.java

```diff
@@ -57,8 +58,11 @@ public static void main(String[] args) throws InterruptedException {
         KafkaProducer producer = new KafkaProducer(props);
         log.info("Sending {} messages ...", config.getMessageCount());
 
+        // Address
         boolean blockProducer = System.getenv("BLOCKING_PRODUCER") != null;
-        boolean transactionalProducer = config.getAdditionalConfig().contains("transactional.id");
+      //  boolean transactionalProducer = !Boolean.parseBoolean(props.getProperty(TRANSACTIONAL_ID_CONFIG));
+        boolean transactionalProducer = config.getProperties().contains("transactional.id");
+      //  boolean transactionalProducer = config.getAdditionalConfig().contains("transactional.id");
```

Comment:
Don't forget to clean up

Reply:
Reverted to before refactoring ConsumerConfig